src/app/components/book-form/book-form.component.ts

```diff
@@ -9,19 +16,62 @@ import { FormBuilder, FormGroup } from '@angular/forms';
 })
 export class BookFormComponent implements OnInit {
 
-  myForm: FormGroup;
-  ratingValues: number[] = [1,2,3,4,5];
+    @Input() action: FormActions;
+    @Input() dialogRef?: MatDialogRef<EditDialogComponent | AddDialogComponent>;
+    @Input() bookToUpdate?: ReadBook;
+    @Input() dialogClosed: EventEmitter<Boolean>;
 
-  constructor(private fb: FormBuilder) { }
+    @Output() updatedBook? = new EventEmitter<Partial<ReadBook>>(); 
+    @Output() newBook? = new EventEmitter<ReadBook>(); 
+    @Output() formInvalid? = new EventEmitter<Boolean>();
 
-  ngOnInit() {
-      this.myForm = this.fb.group({
-          title: '',
-          author: '',
-          rating: '',
-      });
+    ratingValues: number[] = [1,2,3,4,5];  
+    myForm: FormGroup;
+ 
+    constructor(private fb: FormBuilder, private booksService:BooksService){
+    }
 
-      this.myForm.valueChanges.subscribe(console.log);
-  }
+    ngOnInit(){
+        if (this.action === FormActions.Add){
```

Comment:
Looks like switch/case would be more meaningful here.

Reply:
Also, assignments like this should be in constructor.
In this case we can even avoid formBuilder as a field.

Reply:
I can't do it in the constructor because I'm using @Input action which is not available there